src/phoenix/experimental/callbacks/langchain_tracer.py

```diff
@@ -74,10 +82,52 @@ def _prompt_template(serialized: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
                 yield LLM_PROMPT_TEMPLATE_VARIABLES, kwargs.get("input_variables", [])
                 yield LLM_PROMPT_TEMPLATE_VERSION, "unknown"
                 break
-        except TypeError:
+        except (AttributeError, KeyError, TypeError):
             continue
 
 
+def _invocation_parameters(run_extra: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yields invocation parameters if present."""
+    yield LLM_INVOCATION_PARAMETERS, json.dumps(run_extra.get("invocation_params", {}))
+
+
+def _model_name(run_extra: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    """Yields model name if present."""
+    for key in ["model_name", "model"]:
+        try:
+            yield LLM_MODEL_NAME, run_extra["invocation_params"][key]
+            break
+        except (KeyError, TypeError):
+            continue
+
+
+def _token_counts(run_outputs: Dict[str, Any]) -> Iterator[Tuple[str, int]]:
+    """Yields token count information if present."""
+    for attribute_name, key in [
+        (LLM_TOKEN_COUNT_PROMPT, "prompt_tokens"),
+        (LLM_TOKEN_COUNT_COMPLETION, "completion_tokens"),
+        (LLM_TOKEN_COUNT_TOTAL, "total_tokens"),
+    ]:
+        try:
+            yield attribute_name, run_outputs["llm_output"]["token_usage"][key]
+        except (KeyError, TypeError):
+            continue
```

Comment:
nit. put it in a local variable `token_usage`, so it doesn't need to `try` multiple times
```suggestion
        if (attribute_value := token_usage.get(key)):
            yield attribute_name, attribute_value
```

Reply:
addressed in [e8d52cc](https://github.com/Arize-ai/phoenix/pull/1135/commits/e8d52ccc06386b1ea2009f6d2364561b3506cb9a)